apps/framework-cli/src/infrastructure/olap/clickhouse/inserter.rs

```diff
@@ -120,6 +121,59 @@ impl<C: ClickHouseClientTrait + 'static> Inserter<C> {
         Ok(())
     }
 
+    async fn get_front_batch(&self) -> Option<Batch> {
```

Comment:
I think this removes the at least once delivery no?

Can you explain why it doesn't?

The thought is that if another process claims the queue in parallel and that batch goes through then if the next batch gets committed then the intermediary batch won't get replayed.

Reply:
i think there's only one `spawn`ed `task` doing the `flush`

Reply:
Feels like if we merge this we need to a test that shows we don't drop data since that's the current implementation just got

Reply:
Could be a manual test on your laptop where you send a bunch of things kill the process, restart and see records not get dropped